Build specific versions of vscode

dleatherman-civis opened the pull request:
## Description

Required: Please provide a brief description of what this pull request is trying to accomplish.

> We want to introduce copilot as a part of the default code-server experience. Unfortunately, the version of the copilot-chat extension is strongly coupled with the vscode version. In order to avoid breaking copilot-chat whenever we update our code-server version, this takes advantage of existing configuration to build a specific version of code-server. 

### Testing the previous code-server version

This screencap tested a build (with copilot extensions) using the prior version of code-server (4.103.2). 

<img width="2087" height="975" alt="image" src="https://github.com/user-attachments/assets/560ab1d9-6744-475f-9f81-06286be2b2ce" />

### Testing the new code-server version
The code-server version was updated in https://github.com/civisanalytics/docker-code-server/pull/11/commits/4be679435c3fb228d1450965a1d5f09da4e2daff to the current version (4.104.3), along with updates to the copilot and copilot-chat extension versions.
<img width="2082" height="966" alt="image" src="https://github.com/user-attachments/assets/cf0c5b4a-89f4-4458-9aba-0643d5547072" />

## Context, Consequences, & Considerations

Required: Please step through the following list, pausing at each item to consider your change in relation to the item's context.
Check the box to mark that it applies, and enter your relevant notes under the item.

- [ ] Security: This has security implications. This includes (but not limited to) adding users, modifying user/app permissions, network rules/policies, changing a system interconnection, or changing an authorization strategy.
  - [X] This PR does not require security review. These changes are part of a project plan that has already undergone security review. The link is provided below.
  - [ ] This PR requires security review. Add the `security` label to this PR then request a review from the [Security Code Reviewers Team](https://github.com/orgs/civisanalytics/teams/security-code-reviewers).

>

- [ ] Execution: This change requires commands to be run outside of the normal merge.

>

- [ ] Impact: This change may cause service interruptions.

>

- [x] Testing: How did you test this change (unit tests, acceptance tests, etc.)? Did you do any manual testing?

> I changed the base image of civis-vscode, deployed this on a new code cloud, and interacted with the chat agent.

- [x] Testing: How will you confirm this change once it's merged?

> I will do the same.

- [x] Documentation: Documentation to reflect this change has been added to Confluence or Zendesk.

>

- [x] **All items of the checklist have been considered and this PR description is complete.**

dleatherman-civis committed:
REF set code version to older version.

dleatherman-civis committed:
BLD fix the version of code server.

dleatherman-civis committed:
BLD debugging

dleatherman-civis committed:
BLD this fixes it.

dleatherman-civis committed:
REF restore this

dleatherman-civis committed:
DOC add comment

leanne73 reviewed (commented):
I'm a little worried about management of the new variable - unless `CODE_RELEASE` is a magic variable which pre-existing parts of the code use or something?

dleatherman-civis committed:
BLD remove CODE_RELEASE value in version naming & bump version.

This makes the versioning scheme more complicated than it needs to be.
We can increment major versions when updating the version of vscode and
maintain backwards compatibility.

gskifstad commented on buildspec/release.yaml:
```diff
@@ -7,16 +7,18 @@ phases:
+        --build-arg CODE_RELEASE=${CODE_RELEASE}
+        --tag ${FIPS_REPOSITORY_URI}:${CODE_RELEASE}-${PATCH_TAG}
+        --tag ${FIPS_REPOSITORY_URI}:${CODE_RELEASE}-${MINOR_TAG}
+        --tag ${FIPS_REPOSITORY_URI}:${CODE_RELEASE}-${MAJOR_TAG}
```
oh interesting, i guess i had thought it would be something like
* 4.103.2-1.0.0 // first civis release of 4.103.2
* 4.103.2-1.1.0 // updated civis release of 4.103.2
* 4.104.0-1.0.0 // first civis release of 4.104.0

i also was imagining `civis` in there (e.g. `4.104.0-civis1.0.0`), but that is quite long... Could tags use `:`? Not sure if this is more readable: `4.104.0:civis-1.0.0`

I definitely like ` --tag ${FIPS_REPOSITORY_URI}:${CODE_RELEASE}-latest` for the master merge at least, though to our point yesterday, that doesn't help the "you need to make a fake change in civis-vscode to build with the latest changes, e.g. when going 4.103.2-1.0.0  -> 4.103.2-1.1.0"

leanne73 replied:
> In this case, Google AI is incorrect

booo google AI. :sigh:

I suppose another option could be to set our release versions to be equivalent to the related vscode version. This would have the downside of tying us to one release per vscode version though (unless we update the patch version or something else). 

Of all the options, I still lean slightly towards using their version + a tag for us, e.g. `4.104.0-1`; I think this would work with the current buildspec well enough that it wouldn't need any changes (eg. it'd build `4`, `4.104`, and `4.104.0-1`). But I'm also fine with us using our own separate versioning (e.g. `1.0.0`)

gskifstad reviewed (commented):
one suggestion about updating the dev build process, and commented on the tag thread

dleatherman-civis committed:
BLD use CODE_RELEASE in release version docker tag

dleatherman-civis committed:
BLD PR feedback

leanne73 reviewed (approved):
LGTM!
